front_end/src/app/(campaigns-registration)/bridgewater/components/registration-status.tsx

```diff
@@ -0,0 +1,260 @@
+"use client";
+
+import React, { FC, PropsWithChildren, useState } from "react";
+import { BWRegisterButton, ChoicesButtons } from "./hero-section";
+import cn from "@/utils/cn";
+import { CurrentUser } from "@/types/users";
+import BaseModal from "@/components/base_modal";
+import {
+  RegistrationAndSignupForm,
+  RegistrationForm,
+} from "./registration-forms";
+import { CAMPAIGN_KEY } from "../constants";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faCheckCircle } from "@fortawesome/free-solid-svg-icons";
+import { faXmarkCircle } from "@fortawesome/free-regular-svg-icons";
+import { useRouter } from "next/navigation";
+import { useModal } from "@/contexts/modal_context";
+import Link from "next/link";
+
+interface RegisterAndStatusProps {
+  className?: string;
+  currentUser: CurrentUser | null;
+}
+
+export const RegisterAndStatus: FC<RegisterAndStatusProps> = ({
+  className,
+  currentUser,
+}) => {
+  const [dialogState, setDialogState] = useState<{
+    visible: boolean;
+    content: React.ReactNode;
+  }>({ visible: false, content: null });
+
+  const router = useRouter();
+  const [signupCompleteEmail, setSignupCompleteEmail] = useState<string | null>(
+    null
+  );
+
+  const dismissDialog = () =>
+    setDialogState({
+      visible: false,
+      content: null,
+    });
+
+  const HeadingText: FC<{ heading: string; subheading: string }> = ({
+    heading,
+    subheading,
+  }) => {
+    return (
+      <div>
+        <h2 className="my-0 text-base text-gray-0 dark:text-gray-0-dark xs:text-lg sm:text-lg md:text-2xl  xl:text-3xl">
+          {heading}
+        </h2>
+        <p className="mb-0 mt-5 text-sm text-gray-0 dark:text-gray-0-dark xs:text-base sm:text-sm md:text-lg ">
+          {subheading}
+        </p>
+      </div>
+    );
+  };
```

Comment:
Creating components inside parent component is anti-pattern and can lead to performance issues. Let's move it outside. 

The same applies to other internal components, like `LoggedInNotRegisteredFragment`, `EligibilityBox`, etc.

Reply:
Fair point, will take them out at the top level.